oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+<https://github.com/overhangio/tutor/blob/master/tutor/commands/local.py#L91>`__.  This, however, `does not apply to
+Kubernetes support <https://github.com/overhangio/tutor/blob/master/tutor/commands/k8s.py#L281-L288>`__.  While the
+corresponding single-server upgrade procedure can be used as an authoritative reference for how to achieve it manually
+on multi-server deployments, ideally there would be specific documentation on it.
+
+It is important to note that Tutor does not share any code with ``edx/configuration``, nor was it built to update
+containers in place with tools such as Ansible. As such, like other modern container deployment solutions, there is no
+way to update code other than first updating the container image then rolling it out.
+
+
+Criterion 4. Support for master
+...............................
+
+Regis Behmo `reports <https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/5>`__ that some work in
+this direction has already `been done <https://github.com/overhangio/tutor/tree/edge>`__, but at the time of writing it
+is a couple of months behind the latest version of Tutor.
+
+This is clearly an area where contributions are needed. It will likely require one or more volunteers to help maintain
+master and fix breakages, as those would inevitably happen. Continuous deployment and some level of integration testing
+would also be required, as the volunteer(s) would need to be warned as soon as possible about failures.
+
+Close cooperation with the several engineering teams at edX would also need to be campaigned for: it would be best to
+have as much advance warning as possible of backward-incompatible changes coming downstream.
+
+
+Criterion 5. Customization
+..........................
+
+Tutor allows not only using `a custom Docker registry and images
+<https://docs.tutor.overhang.io/configuration.html#custom-images>`__, but also `building custom images
+<https://docs.tutor.overhang.io/configuration.html#custom-open-edx-docker-image>`__.  The latter is how one adds custom
+themes, xblocks, or even uses `a fork of edx-platform
+<https://docs.tutor.overhang.io/configuration.html#running-a-fork-of-edx-platform>`__.  In operational terms it is no
+harder than how one would accomplish the same with ``edx/configuration``.
+
+For many things that Tutor doesn't explicitly support (and for which there is no desire/need to include in the core), it
+is possible to `write plugins <https://docs.tutor.overhang.io/plugins/gettingstarted.html>`__.  With this engine, one
+can add additional services and configure the rest of the platform to use them. One can imagine writing plugins that
+allow deployment of the different Open edX MFEs, for instance.
+
+A note on theming: it would be ideal if it were possible to store static assets separately so that a single base image
+could be used for multiple instances, but that is not currently the case.
```

Comment:
I understand your point. This is a decision that was made [early on](https://github.com/overhangio/tutor/pull/102). The idea was that when we ship an image it should already contain everything we need (outside of settings) to get running. On the other hand, you want to be able to build images quickly, even when they bundle different themes.

As the static asset compilation is the most complex one for the `openedx` Docker image, it is performed last in the Dockerfile. This means that you should be able to re-build the image fairly quickly for different themes, by leveraging Docker layer caching.

If you need to ship a single image for all platforms, then I see only two options, none of which are great:

1. Bundle all themes inside a single image.
2. Mount different volumes in `/openedx/staticfiles` at runtime.

Reply:
@regisb, I think the best idea so far was @fghaas': turning to S3 for this, maybe as a Tutor plugin, will probably be the simplest way forward.